extensions/cxf-soap/runtime/pom.xml

```diff
@@ -0,0 +1,136 @@
+    this work for additional information regarding copyright ownership.
+    The ASF licenses this file to You under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with
+    the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.camel.quarkus</groupId>
+        <artifactId>camel-quarkus-cxf-soap-parent</artifactId>
+        <version>2.11.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>camel-quarkus-cxf-soap</artifactId>
+    <name>Camel Quarkus :: CXF :: Runtime</name>
+    <description>Expose SOAP WebServices using Apache CXF or connect to external WebServices using CXF WS client.</description>
+
+    <properties>
+        <camel.quarkus.jvmSince>2.11.0</camel.quarkus.jvmSince>
+        <camel.quarkus.nativeSince>2.11.0</camel.quarkus.nativeSince>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-cxf-soap</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-cxf-common</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-cxf-transport</artifactId>
+        </dependency>
+
+        <!-- CXF WS-* dependencies -->
+        <dependency>
+            <groupId>org.apache.cxf</groupId>
+            <artifactId>cxf-rt-ws-rm</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.cxf</groupId>
+            <artifactId>cxf-rt-ws-addr</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.cxf</groupId>
+            <artifactId>cxf-rt-ws-security</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.cxf</groupId>
+            <artifactId>cxf-rt-ws-policy</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.cxf</groupId>
+            <artifactId>cxf-rt-transports-http-jetty</artifactId>
```

Comment:
This is for the WS server part, right?

If so, similar to my previous comment, I don't think this should be a dependency of the extension by default. So it should be moved to the BOM.

However, I do not want us to be supporting HTTP transports like Jetty. We should be integrating with the Quarkus Vert.x HTTP server.

Reply:
Correct, I can move it out. I agree that ideally we want to support Vert.x, but CXF doesn't provide transport implementation for it (and I'm not sure if they will). I can try to implement one, but it may take a while. So as an interim solution let's support it and I'll see what I can do about supporting other transports.

Reply:
I created a ticket here: https://issues.apache.org/jira/browse/CXF-8742
Please feel free to add your comments and/or corrections.

Reply:
To me it's mandatory that the endpoints are run on the Quarkus Vert.x HTTP server. Otherwise we're just making the app performance worse and exposing ourselves to more CVEs.

AFAIK `quarkiverse-cxf` has already solved this problem.

Reply:
Yes, I tried it, for some reason CXF does not recognize the VertxDestinationFactory that is provided by quarkiverse-cxf. I'll keep investigating.

Reply:
So maybe we can consider to leverage on [quarkus-cxf](https://github.com/quarkiverse/quarkus-cxf).

Reply:
I'm trying to figure out how it's done there, but it seems like we'd have to rewrite Camel CXF completely, because there seems to be no way to pass the factories defined in the quarkus-cxf to Camel...

Reply:
OK, I think I know what's going on - currently Camel does not allow creating custom JaxWsServerFactroryBean: https://github.com/apache/camel/blob/main/components/camel-cxf/camel-cxf-soap/src/main/java/org/apache/camel/component/cxf/jaxws/CxfEndpoint.java#L683
This means that destination factories are initialized in the Camel CXF code, and apparently custom destination factories are ignored. Not sure if there's any way around it, apparently not...

Reply:
@ffang can you take a look?